synchronization/pipeline.h

```diff
@@ -133,6 +135,7 @@ class Pipeline : public fml::RefCountedThreadSafe<Pipeline<R>> {
     empty_.Signal();
 
     TRACE_FLOW_END("flutter", "PipelineItem", trace_id);
+    TRACE_EVENT_ASYNC_END0("flutter", "PipelineItem", trace_id);
```

Comment:
Just curious: why are there two `TRACE_EVENT_ASYNC_END0("flutter", "PipelineItem", trace_id)` (one here ,one in ` ~ProducerContinuation()`), but only one `TRACE_EVENT_ASYNC_BEGIN0("flutter", "PipelineItem", trace_id_)` in the `ProducerContinuation` constructor? Could it be possible that BEGIN is called once while END is called twice? Maybe it's Ok to call END multiple times?

Reply:
A trace can end only once. The end call in the destructor is if the pipeline drops the continuation (while shutting down the engine maybe). See the `if (continuation_)`.